service/history/workflow/mutable_state.go

```diff
@@ -201,6 +203,7 @@ type (
 		IsTransientWorkflowTask() bool
 		ClearTransientWorkflowTask() error
 		HasBufferedEvents() bool
+		HasAnyBufferedEvent(filter BufferedEventFilter) bool
```

Comment:
I didn't see the caller of this method on my phone screen. Could you point me to the caller?

Reply:
it's used in workflowTaskHandlerCallback.go to see if any bufferedEvent needs a workflow task